azurerm_container_app: secret.name validation has wrong error output

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.8

### AzureRM Provider Version

4.8.0

### Affected Resource(s)/Data Source(s)

azurerm_container_app

### Terraform Configuration Files

```hcl
resource "azurerm_container_app" "example" {
  name                         = "example-app"
  container_app_environment_id = azurerm_container_app_environment.example.id
  resource_group_name          = azurerm_resource_group.example.name
  revision_mode                = "Single"

  secret {
    name  = "this.is.an.invalid.secret.name"
    value = "secret"
  }

  template {
    container {
      name   = "examplecontainerapp"
      image  = "mcr.microsoft.com/k8se/quickstart:latest"
      cpu    = 0.25
      memory = "0.5Gi"
    }
  }
}
```


### Debug Output/Panic Output

```shell
2024-11-07T10:48:47.648+0100 [ERROR] vertex "module.redacted.main" error: "secret.0.name" must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
2024-11-07T10:48:47.648+0100 [WARN]  provider.terraform-provider-azurerm_v4.8.0_x5: [WARN] Truncating attribute path of 1 diagnostics for TypeSet
2024-11-07T10:48:47.648+0100 [ERROR] provider.terraform-provider-azurerm_v4.8.0_x5: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_attribute="AttributeName(\"secret\")" tf_proto_version=5.6 tf_req_id=2e265cd9-ec49-8f32-d98d-68960eb4aede diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="\"secret.0.name\" must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character" @module=sdk.proto tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig timestamp="2024-11-07T10:48:47.648+0100"
2024-11-07T10:48:47.648+0100 [ERROR] vertex "module.redacted.main" error: "secret.0.name" must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
╷
│ Error: "secret.0.name" must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character
│ 
│   with module.redacted.main,
│   on modules/redacted/main.tf line 17, in resource "azurerm_container_app" "main":
│   17: resource "azurerm_container_app" "main" {
│ 
```


### Expected Behaviour

Allowance of '.' in `azurerm_container_app.secret.name`

### Actual Behaviour

'.' is not allowed. The Azure portal UI is incoherent here (see pictures) but you are not allowed to create secrets in container app containing '.' in name/key. Also max length is not validated in provider.

![Screenshot 2024-11-07 at 10 31 10](https://github.com/user-attachments/assets/2495d4f2-046a-4ba5-8a15-0a0baf351ad7)
![Screenshot 2024-11-07 at 10 31 21](https://github.com/user-attachments/assets/34ad6721-d929-431d-8584-723927ce4d9a)


### Steps to Reproduce

`terraform plan`

### Important Factoids

_No response_

### References

Created a PR: https://github.com/hashicorp/terraform-provider-azurerm/pull/27935

--- a/internal/services/containerapps/validate/validate.go
+++ b/internal/services/containerapps/validate/validate.go
@@ -46,8 +46,8 @@ func SecretName(i interface{}, k string) (warnings []string, errors []error) {
 		return
 	}
 
-	if matched := regexp.MustCompile(`^[a-z0-9][a-z0-9-]*[a-z0-9]?$`).Match([]byte(v)); !matched || strings.HasSuffix(v, "-") || strings.HasSuffix(v, ".") {
-		errors = append(errors, fmt.Errorf("%q must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character", k))
+	if matched := regexp.MustCompile(`^[a-z0-9][a-z0-9-]{0,251}[a-z0-9]?$`).Match([]byte(v)); !matched || strings.HasSuffix(v, "-") {
+		errors = append(errors, fmt.Errorf("%q must consist of lower case alphanumeric characters, '-', and must start and end with an alphanumeric character. The length must not be more than 253 characters", k))
 	}
 	return
 }

--- a/internal/services/containerapps/validate/validate_test.go
+++ b/internal/services/containerapps/validate/validate_test.go
@@ -99,6 +99,10 @@ func TestValidateSecretNames(t *testing.T) {
 			Valid: false,
 		},
 		{
+			Input: "no.dots.allowed",
+			Valid: false,
+		},
+		{
 			Input: "a",
 			Valid: true,
 		},

